server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTarget.java

```diff
@@ -170,7 +171,21 @@ public void startReplication(ActionListener<Void> listener) {
             final List<StoreFileMetadata> filesToFetch = getFiles(checkpointInfo);
             state.setStage(SegmentReplicationState.Stage.GET_FILES);
             cancellableThreads.checkForCancel();
-            source.getSegmentFiles(getId(), checkpointInfo.getCheckpoint(), filesToFetch, indexShard, getFilesListener);
+            final BiConsumer<String, Long> fileProgressTracker = (fileName, bytesRecovered) -> {
+                ReplicationLuceneIndex index = state.getIndex();
+                if (index != null && index.getFileDetails(fileName) != null) {
+                    index.addRecoveredBytesToFile(fileName, bytesRecovered);
```

Comment:
ReplicationLuceneIndex is a thread safe class (all methods are synchronized). However, the "check if present then add bytes" pattern is not thread safe because the file could be removed in between the check and the add.

Reply:
Good point. In that case, we do have the scenario covered with proper usage - the files are calculated and fetched [here](https://github.com/kotwanikunal/OpenSearch/blob/main/server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTarget.java#L222-L225) and the same list is passed down for downloads [here](https://github.com/kotwanikunal/OpenSearch/blob/main/server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTarget.java#L170-L173). 

And any other incorrect update will lead to an exception - which I think is valid and points to a bug in the code.
Removing the contains check.